cmd/controller-manager/main.go

```diff
@@ -84,10 +92,28 @@ func main() {
 		}
 	}
 
-	// TODO(marun) Make configurable and default to a namespace provided via the downward api
-	fedNamespace := util.DefaultFederationSystemNamespace
-	clusterNamespace := util.MulticlusterPublicNamespace
+	// Allow federation and cluster namespaces to be overridden by
+	// environment variables to allow the namespaces to be set to the
+	// pod namespace via the downward api.
+	fedNamespaceFromEnv := os.Getenv("FEDERATION_NAMESPACE")
```

Comment:
I'm curious why we need to make this controller aware of an environment variable name when we can set the CLI flags in the pod spec. You can inject from downward API envs into the entrypoint and command fields of the pods. I don't think that should block this PR, though.

Reply:
That's just my ignorance - I missed the detail that env vars could be used directly in the args.  I suggest fixing in a follow-up.

Reply:
Yeah, no need to block this PR at all.